controllers/argocd/route.go

```diff
@@ -366,3 +381,35 @@ func (r *ReconcileArgoCD) reconcileApplicationSetControllerWebhookRoute(cr *argo
 	}
 	return r.Client.Update(context.TODO(), route)
 }
+
+func shortenHostname(hostname string) (string, error) {
```

Comment:
This appears not to be implementing the full algorithm prescribed, specifically bullet point 3:
After all the labels are 63 characters or less, check the length of the overall hostname:
- If the overall hostname is > 255, then shorten the FIRST label until the host name is < 255
- After the FIRST label has been shortened, if it is < 20, then return an error (this is a sanity test to ensure the label is likely to be unique)